docs/design/proposals/logical-volume-manager-storage-api-v2.md

```diff
@@ -0,0 +1,591 @@
+the Node contains more detailed conditions that allow setting up detailed status checks
+
+#### `LVMCluster` replaced by `LVMNodeSet`
+
+The current LVMCluster will be broken up into node-specific configurations, wherein each configuration is created through the `LVMNodeSet`,
+which contains a few key spec elements:
+
+1. `spec.nodeSelector` to determine on which nodes to create LVM configurations
+2. `spec.template.volumeGroups` to determine which volumeGroups to create and with which devices based on a `deviceSelector`
+3. `spec.template.deviceClasses` to determine which [TopoLVM device classes](https://github.com/topolvm/topolvm/blob/main/docs/lvmd.md#lvmd) should be prepared and configured for lvm.
+   This also configures a corresponding StorageClass and SnapshotClass for that deviceClass.
+
+```yaml
+---
+apiVersion: lvm.topolvm.io/v2alpha1
+kind: LVMNodeSet
+metadata:
+  name: my-lvmnodeset
+  uid: 31167fec-f8d3-47af-883f-05aaa153da88
+spec:
+  nodeSelector:
+    nodeSelectorTerms:
+      - matchExpressions:
+          - key: topology.kubernetes.io/zone
+            operator: In
+            values:
+              - antarctica-east1
+              - antarctica-west1
+      - matchExpressions:
+          - key: custom.topology.domain.io/storage
+            operator: In
+            values:
+              - archive-storage
+  template:
+    volumeGroups:
+      - name: vg1
+        deviceSelector:
+          deviceSelectorTerms:
+            - matchExpressions:
+                - key: kname
+                  operator: In
+                  values:
+                    - "/dev/mapper/some-encrypted-device"
+                    - "/dev/mapper/some-fallback-device"
+            - matchExpressions:
+                - key: kname
+                  operator: In
+                  values:
+                    - "/dev/mapper/some-default-device"
+      - name: othervg
+        deviceSelector:
+          deviceSelectorTerms:
+            - matchExpressions:
+                - key: kname
+                  operator: Any
+                - key: size
+                  operator: GreaterThan
+                  values:
+                    - "30Gi"
+      - name: specific-serials
+        deviceSelector:
+          deviceSelectorTerms:
+            - matchExpressions:
+                - key: serial
+                  operator: In
+                  values:
+                    - "2313AW459310"
+    deviceClasses:
+      - name: thinly-provisioned
+        strategy: Thin
+        volumeGroup: vg1
+        default: true
+        thin:
+          poolName: thin-pool-1
+          sizePercent: 90
+          overprovisionRatio: 10
+        storageClass:
+          namingPolicy: Prefixed
+          prefix: "lvms"
+        snapshotClass:
+          namingPolicy: Prefixed
+          prefix: "lvms"
+status:
+  state: Ready
+  nodes:
+    - name: archive-storage-node-in-antarctica-east1
+      state: Ready
```

Comment:
Would adding a `message` field here make sense to increase our flexibility?

Reply:
I thought of the same but didn't do this as it's common to only relay the single source of truth to the managing resource and have all details hidden in the other CRD. of course this can be changed later on to add more fields if necessary since its just a mapping.